src/Microsoft.OData.Core/SelectedPropertiesNode.cs

```diff
@@ -469,6 +583,76 @@ private static IEnumerable<string> GetPossibleMatchesForSelectedOperation(IEdmOp
             yield return qualifiedContainerName + operationNameWithParameters;
         }
 
+        private bool IsValidExpandToken(string item)
+        {
+            // Check #1: Expand token must contain '(' and end with ')', must be related to navigation property.
+            int idxLP = item.IndexOf('(');
+            if (idxLP == -1
+                || !item.EndsWith(")", StringComparison.Ordinal)
+                || (this.structuredType != null && IsBoundOperationToken(item.Substring(0, idxLP))))
+            {
+                // selected item is not properly parenthesized, or is an operation token.
+                return false;
+            }
+
+            // Check #2: parentheses must be balanced.
+            int count = 0;
+            foreach (char c in item)
+            {
+                if (c == '(')
+                {
+                    ++count;
+                }
+                else if (c == ')')
+                {
+                    --count;
+                }
+
+                if (count < 0)
+                {
+                    // Malformed parentheses detected ==> not expand token.
+                    return false;
+                }
+            }
+
+            return count == 0;
+        }
+
+        /// <summary>
+        /// Checks the specified model and structuredType and see whether the token can be resolved to a bound operation name.
+        /// </summary>
+        /// <param name="token">The token to check.</param>
+        /// <returns>true if token can be matched a bound operation; false otherwise.</returns>
+        private bool IsBoundOperationToken(string token)
+        {
+            const char nameSpaceSeparator = '.';
+
+            if (this.edmModel == null)
+            {
+                return false;
+            }
+
+            bool found = false;
+            int idx = token.LastIndexOf(nameSpaceSeparator);
+            if (idx == -1)
+            {
+                // Token is in unqualified form, try matching the unqualified name of the bound operation.
+                // N.B.: Operation name should be always name space qualified per current OData spec. The older
+                // version of the library returns unqualified. We should fix the library such that it no longer
+                // returns unqualified names in context url.
+                found = this.edmModel.FindDeclaredBoundOperations(this.structuredType)
+                        .Any(op => op.Name.Equals(token, StringComparison.Ordinal));
+            }
+            else
+            {
+                // Qualified name: try to match bound and unbound operations
+                found = this.edmModel.FindBoundOperations(token, this.structuredType).Count() != 0
+                    || this.edmModel.FindDeclaredOperations(token).Count() != 0;

```

Comment:
>|| this.edmModel.FindDeclaredOperations(token).Count() != 0 [](start = 20, length = 59)

Why is this part there? If the operation is not bound to this type, then we don't care about it here.